Sources/Utility/ArrayExtensions.swift

```diff
@@ -20,10 +20,10 @@ extension Array {
         var t = [T]()
         var u = [U]()
         for e in self {
-            if e is T {
-                t.append(e as! T)
-            } else {
-                u.append(e as! U)
+            if let e = e as? T {
+                t.append(e)
+            } else if let e = e as? U {
+                u.append(e)
```

Comment:
@aciidb0mb3r's solution is syntactic sugar (https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Patterns.html#//apple_ref/swift/grammar/optional-pattern). Same solution, different way of implementing, which I would deem non-traditional.